src/main/java/org/logstash/dissect/fields/FieldFactory.java

```diff
@@ -0,0 +1,37 @@
+package org.logstash.dissect.fields;
+
+public final class FieldFactory {
+    private static final String MIXED_PREFIXES = "Field cannot prefix with both Append and Indirect Prefix (%s): %s";
+    private static final String PREFIXED_EMPTY = "Field cannot be a prefix on its own without further text";
+
+    public static Field create(String field) {
+        if (field.isEmpty() || field.startsWith("?")) {
```

Comment:
This class contains tokenization logic and the field creation code. I still think it pays off to have a dedicated parser but maybe that is just me. :) For me it is hard to follow all the grammar rules in these different classes (as you might infer from my questions above which always ask whether the enclosing code guarantees this and that...)